File: 백준/Silver/2822. 점수 계산/점수 계산.py
#참가자의 총 점수는 가장 높은 점수 5개의 합
score = [0]*8
sum  = 0
sum =0
get_score=[]
get = []
for i in range(8):
    score = int(input())
    get_score.append((score,i+1))
get_score.sort()
for i in range(5):
    sum += get_score[::-1][i][0]
    get.append(get_score[::-1][i][1])
get.sort()
print(sum)
print(*get)

File: 백준/Silver/2822. 점수 계산/점수 계산.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()
	var recToMap = map[int]int{}
	var record = make([]int, 8)
	var get = make([]int, 5)
	var sum int

	for i := 0; i < 8; i++ {
		var num int
		fmt.Fscanf(reader, "%d\n", &num)
		recToMap[num] = i
		record[i] = num
	}
	sort.Ints(record)
	for i := 1; i <= 5; i++ {
		get[i-1] = recToMap[record[8-i]]
		sum += record[8-i]
	}
	sort.Ints(get)
	fmt.Fprintf(writer, "%d\n", sum)
	for _, v := range get {
		fmt.Fprintf(writer, "%d ", v+1)
	}
}
